packages/core/src/storages/sitemap_request_list.ts

```diff
@@ -418,7 +418,7 @@ export class SitemapRequestList implements IRequestList {
         void requestList.load({ parseSitemapOptions: options.parseSitemapOptions });
 
         if (requestList.persistenceOptions.enable) {
-            requestList.events.on(EventType.PERSIST_STATE, requestList.persistState);
+            requestList.events.on(EventType.PERSIST_STATE, async () => await requestList.persistState());
```

Comment:
if you do it like this then the deregistering wont work, right? 

i would go with saving the reference to the bound function